pyomo/solvers/tests/checks/test_CBCplugin.py

```diff
@@ -0,0 +1,314 @@
+        self.model.X = Var(within=NonNegativeIntegers)
+        self.model.C1 = Constraint(expr=self.model.X <= 1)
+        self.model.C2 = Constraint(expr=self.model.X >= 2)
+        self.model.Obj = Objective(expr=self.model.X, sense=minimize)
+
+        results = self.opt.solve(self.model)
+
+        self.assertEqual(ProblemSense.minimize, results.problem.sense)
+        self.assertEqual(TerminationCondition.infeasible, results.solver.termination_condition)
+        self.assertEqual('Model was proven to be infeasible.', results.solver.termination_message)
+        self.assertEqual(SolverStatus.warning, results.solver.status)
+
+    @unittest.skipIf(not cbc_available, "The 'cbc' solver is not available")
+    def test_unbounded_mip(self):
+        self.model.X = Var(within=Integers)
+        self.model.Obj = Objective(expr=self.model.X, sense=minimize)
+
+        instance = self.model.create_instance()
+        results = self.opt.solve(instance)
+
+        self.assertEqual(ProblemSense.minimize, results.problem.sense)
+        self.assertEqual(TerminationCondition.unbounded, results.solver.termination_condition)
+        self.assertEqual('Model was proven to be unbounded.', results.solver.termination_message)
+        self.assertEqual(SolverStatus.warning, results.solver.status)
+
+    @unittest.skipIf(not cbc_available, "The 'cbc' solver is not available")
+    def test_optimal_mip(self):
+        self.model.Idx = RangeSet(2)
+        self.model.X = Var(self.model.Idx, within=NonNegativeIntegers)
+        self.model.Y = Var(self.model.Idx, within=Binary)
+        self.model.C1 = Constraint(expr=self.model.X[1] == self.model.X[2] + 1)
+        self.model.Obj = Objective(expr=self.model.Y[1] + self.model.Y[2] - self.model.X[1],
+                                   sense=maximize)
+
+        results = self.opt.solve(self.model)
+
+        self.assertEqual(1.0, results.problem.lower_bound)
+        self.assertEqual(1.0, results.problem.upper_bound)
+        self.assertEqual(results.problem.number_of_binary_variables, 2)
+        self.assertEqual(results.problem.number_of_integer_variables, 4)
+        self.assertEqual(ProblemSense.maximize, results.problem.sense)
+        self.assertEqual(TerminationCondition.optimal, results.solver.termination_condition)
+        self.assertEqual(
+            'Model was solved to optimality (subject to tolerances), and an optimal solution is available.',
+            results.solver.termination_message)
+        self.assertEqual(SolverStatus.ok, results.solver.status)
+
+
+class TestCBCUsingMock(unittest.TestCase):
+    """
+    These tests are here to test the way we are parsing soln and log files
+    The files were produced by using the following configuration:
+    macOS Mojave Version 10.14.3
+    CBC Version 2.9.9
+    appdirs==1.4.3
+    certifi==2018.11.29
+    nose==1.3.7
+    numpy==1.16.1
+    ply==3.11
+    Pyomo==5.6.1
+    PyUtilib==5.6.5
+    six==1.12.0
+
+
+    n = 30
```

Comment:
I think I can safely delete the contents of the lp files, which should be fine as there are instructions in the tests to recreate the files.

I'll try and make n smaller to trim down the log files, but I'll probably have to have a look at turning off/down some solver settings (like the feasibility pump) otherwise it may find an optimal solution before producing many of these termination conditions.

Reply:
Note that the `.log` files are not the big consumer of space, but rather the `.lp` and `.soln` files.  I think you can probably leave the solver options as they are.

Reply:
I've found the minimal n (19) that still evaluates some nodes on the B&B tree.
Files are now smaller.